### PR TITLE
OE-299 and OE-303 conflict popup changes

### PR DESCRIPTION
**Pull request by apigram**
Conflict checking will now only run if all three required fields are populated and valid.
A message will appear in the popup if the fields are populated but are not valid, containing all relevant error messages.
Added CERA number to the conflicts table. This possesses the same link to the patient details screen.

### CODE REVIEW

File: protected/views/patient/crud/_conflicts.php [hunk 1]
```diff
@@ -1,47 +1,50 @@
...
+                                            Yii::app()->controller->createUrl('patient/view', array('id' => $patient->id)), array('target' => '_blank')); ?>
+                                    </td>
+                                </tr>
+                            <?php endforeach; ?>
+                            </tbody>
+                        </table>
+                    </div>
+                </div>
+            <?php else: ?>
+                <p>No conflicts found.</p>
```
Comment (marshl) [inline protected/views/patient/crud/_conflicts.php]: I wonder if we should tell the user a bit more. They might wonder exactly what a conflict was, and why this banner appeared. Perhaps we should have some text explaining what you have to fill in for the conflict check summary to appear as well. We could even hide the rest of the fields on the screen until the conflict has been resolved. That could cover the other issue of them ignoring the conflict.

File: protected/models/Patient.php [hunk 2]
```diff
@@ -2089,7 +2089,7 @@ public function dateOfBirthRangeValidator($attribute, $params)
      * @param $last_name string Last name.
      * @param $dob string Date of Birth (DD/MM/YYYY).
      * @param $id int ID of the current patient record.
-     * @return array|Patient[] The list of patients who have similar names and the same date of birth.
+     * @return array The list of patients who have similar names and the same date of birth, or the invalid patient model.
```
Comment (marshl) [inline protected/models/Patient.php]: I know you mentioned this earlier, but I didn't realise that you were actually mixing return types, that feels a bit messy. I'm not sure of a good way around it though.

Comment (apigram) [reply protected/models/Patient.php]: This is the only way I could think of to make this work without a significant refactor.

File: protected/views/patient/crud/_conflicts.php (hunk 1 above)
Comment (apigram) [reply protected/views/patient/crud/_conflicts.php]: Possibly. Ultimately that decision rests with CERA though.

**Review by marshl — APPROVED** [review]

**Pull request merged**